RK3

SCaptainCAP opened the pull request:
Ссылка на игру: https://cyclicgame.herokuapp.com/

SCaptainCAP committed:
Websockets reAdded

algys committed:
Revert: Delete WebSocketServerConfiguration.java

algys committed:
add game mechanics

algys committed:
Revert: Delete ConnectedSessionsService.java

algys committed:
fix test

SCaptainCAP committed:
Socket class layout added

SCaptainCAP committed:
Deleted old files

SCaptainCAP committed:
Some logic update

SCaptainCAP committed:
Game logic update

SCaptainCAP committed:
Game logic update

SCaptainCAP committed:
Added some new api

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Bug fixes

algys committed:
add cors origin for front test

algys committed:
fix cors

algys committed:
add heroku to origins

algys committed:
try to disable httponly cookies

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Update AuthorizedUsersService.java

SCaptainCAP committed:
Update AccountServiceDB.java

SCaptainCAP committed:
Organized imports and changed api

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Added resources

SCaptainCAP committed:
Properties fixed for heroku

SCaptainCAP committed:
Properties fixed for heroku. Again

KSolovyev commented:
Добавьте ссылку на фронт в описание пул реквеста

KSolovyev commented on src/main/java/com/cyclic/configs/Constants.java:
```diff
@@ -0,0 +1,37 @@
+    public static final int DATATYPE_PLAYER_DISCONNECT = 7;
+
+    // ACTIONs are used to send data TO server
+    public static final int ACTION_READY_FOR_ROOM_SEARCH = 1;
```
в enum

KSolovyev commented on src/main/java/com/cyclic/configs/Constants.java:
```diff
@@ -0,0 +1,37 @@
+public class Constants {
+
+    // DATATYPEs are used to send data FROM server
+    public static final int DATATYPE_ROOMINFO = 1;
```
в enum

KSolovyev commented on src/main/java/com/cyclic/configs/Constants.java:
```diff
@@ -0,0 +1,37 @@
+    // ROOM statuses
+    public static final int STATUS_CREATING = 0;
+    public static final int STATUS_PLAYING = 1;
+    public static final int STATUS_FULL_SMB_NOT_READY = 2;
```
здесь вообще все выглядит, как enum'ы

KSolovyev commented on src/main/java/com/cyclic/controllers/WebSocketController.java:
```diff
@@ -0,0 +1,120 @@
+
+    @Override
+    public void handleTransportError(WebSocketSession session, Throwable throwable) throws Exception {
+        //System.out.println("errorConsole occured at sender " + session);
```
комментариев в коде быть не должно

KSolovyev commented on src/main/java/com/cyclic/models/game/GameField.java:
```diff
@@ -0,0 +1,137 @@
+        world = new Node[height][width];
+        for (int i = 0; i < height; i++) {
+            for (int j = 0; j < width; j++) {
+                world[i][j] = null;
```
в java массивы автоматически инициализируются null'ами

SCaptainCAP committed:
Some fixes and repackaging

SCaptainCAP committed:
Fixed heroku path

KSolovyev commented on src/main/java/com/cyclic/models/game/GameField.java:
```diff
@@ -0,0 +1,223 @@
+
+    public void setNodeToPosition(int beginX, int beginY, Node node) {
+        if (beginX < 0 || beginX >= width || beginY < 0 || beginY >= height)
+            return;
```
и все? просто забьем?

KSolovyev commented on src/main/java/com/cyclic/models/game/GameField.java:
```diff
@@ -0,0 +1,223 @@
+        room.broadcast(room.getGson().toJson(new NewBonusBroadcast(bonuses)));
+    }
+
+    public Point findRandomNullPoint() {
```
пометьте как `@Nullable`

KSolovyev commented on src/main/java/com/cyclic/models/game/GameField.java:
```diff
@@ -0,0 +1,223 @@
+    }
+
+    public void stopGame() {
+        room = null;
```
и все?

KSolovyev commented on src/main/java/com/cyclic/models/game/GameField.java:
```diff
@@ -0,0 +1,223 @@
+            if (world[y][x] == null) {
+                return new Point(x, y);
+            }
+        }
```
весь алгоритм выглядит громоздко. Можно лучше

KSolovyev commented on src/main/java/com/cyclic/models/game/GameField.java:
```diff
@@ -0,0 +1,223 @@
+        move.setType(Move.MoveType.ACCEPT_OK);
+    }
+
+    private final static Random randomDelta = new Random(); // TODO check
```
а раньше вы ThreadLocalRandom использовали

SCaptainCAP committed:
Fixed bug with all moves were bad

KSolovyev reviewed (changes requested):
будет еще

SCaptainCAP committed:
A lot of changes
1) Now we don't use int constants - we use enums
2) Move answers are changed a lot. Now we send as Algys said to do it.
3) Some repackaging
4) Bonus is now tower with playerID = -1
5) Node now does not contain its linked nodes - we save memory

SCaptainCAP committed:
A lot of changes
1) New algorithm of searching random null point - now we do not iterate
2) Now server sends to front removed links when we attack enemy
3) Server gives every user's tower count after every move
4) Added highscore to the database
5) Fixed bug when user could see password through some requests

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
A lot of changes:
1) New roomManager. Now you can choose roomCapacity to play with desired players count. New balancer. You cannot be alone forever in the room anymore
2) There is no actioncode of ready to game start. Room starts automatically when it is full
3) There is new ping-pong action code
4) User can now leave room with actioncode

SCaptainCAP committed:
Some bug fixes

SCaptainCAP commented:
Исправлены замечания

SCaptainCAP committed:
Now if user don't move for a long time, the move goes to the next player

KSolovyev commented on src/main/java/com/cyclic/models/game/GameField.java:
```diff
@@ -0,0 +1,363 @@
+            return null;
+        world[node.getY()][node.getX()] = null;
+        if (node.isBonus()) {
+            Vector<RNode> deleteNodes = new Vector<>();
```
а почему здесь вектор?

SCaptainCAP replied:
Потому что метод killNode возвращает специальный класс 
`private class NodesAndLinks`,
содержащий 2 вектора - удаленные вершины и удаленные связи. Эти вектора мы как раз и собираем. Вы имеете ввиду, что лучше заменить их на `ArrayList`?

KSolovyev replied:
Да, Vector практически не используется нигде

SCaptainCAP replied:
Хорошо, изменю на ArrayList

SCaptainCAP replied:
Изменено на ArrayList

KSolovyev commented on src/main/java/com/cyclic/models/game/Room.java:
```diff
@@ -0,0 +1,302 @@
+    private Enums.RoomStatus status;
+    private long roomID;
+
+    private transient int performingPlayerIndex;
```
а зачем помечать их как transient?

SCaptainCAP replied:
Потому что мы используем Gson как сериализатор. Он не сериализует поля, отмеченные `transient`. Так как в последствии мы сериализуем комнату и отправляем ее клиенту, я пометил поля, которые клиенту знать не нужно

KSolovyev commented on src/main/java/com/cyclic/models/game/Room.java:
```diff
@@ -0,0 +1,302 @@
+                else {
+                    moveBroadcast = field.acceptMove(player, move);
+                    if (moveBroadcast == null) {
+                        player.disconnectBadApi("Your are moving like an asshole");
```
не надо так

SCaptainCAP replied:
Пофикшено. Хотя, это ускоряло процесс разработки фронтенда

KSolovyev replied:
я имел в виду сообщение

SCaptainCAP replied:
Так я тоже :)

KSolovyev commented on src/main/java/com/cyclic/models/game/GameField.java:
```diff
@@ -0,0 +1,363 @@
+    }
+
+    @Nullable
+    public MoveBroadcast acceptMove(Player player, Move move) {
```
move-то у вас nullable. NPE чую я здесь. 
А вообще не очень делать таймаут через нулевой ход. Чувствуется здесь архитектурный косяк, который когда понадобится еще что-нибудь заставит вас очень сильно рефакторить код механики

SCaptainCAP replied:
Мы принимаем ходы, отправленные разными игроками. Я посчитал, что Таймер таймаута - это некая сущность, похожая на игрока, который ходит вместо ходящего игрока, но передает "Никакой" ход. Так что это фича. По крайней мере, я не могу придумать ситуаций, в которых это помешает расширению фишек сервера. Для этого кстати достаточно расширить класс `Move`, добавив в него какие-то новые поля, и парся их уже после проверки хода на null

KSolovyev replied:
А он может передавать специльный ход, который передает управление следующему игроку?

SCaptainCAP replied:
Да, было бы логично. Добавлю сегодня такую реализацию

SCaptainCAP replied:
Реализовано с помощью наследования

SCaptainCAP committed:
Now we can Win or lose, and leave room without reconnection

SCaptainCAP committed:
Fixed pull request promises

KSolovyev commented:
у меня не получается зарегистрироваться

KSolovyev commented:
А, это был ник короткий. Вообще не понятно

KSolovyev commented:
А как передать ход другому игроку?

SCaptainCAP commented:
Мы выложим сегодня новую версию на Хероку. Та очень сильно устарела. Ошибки логина-регистрации мы уже показываем на новом фронте

KSolovyev commented:
отпишитесь, как сделаете

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Fixed Nullable annotations

SCaptainCAP committed:
Fixed tests

algys committed:
added game url

SCaptainCAP committed:
Update application.properties

Reverted app.properties for heroku

SCaptainCAP committed:
Fixed bug when player, who leaved room and tryed to play in another one could get phantom points in server responses

SCaptainCAP committed:
Added heroku unsafe http origin

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Bug fixes or not fixes

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Added guest sessions; Now you cannot spawn too close

SCaptainCAP committed:
Fixed tests

SCaptainCAP committed:
Update application.properties

SCaptainCAP committed:
Logic changed. Now you can spectate room

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Merge branch 'dev-new-logic' into dev

# Conflicts:
#	src/main/resources/application.properties

SCaptainCAP committed:
Bug fixes

SCaptainCAP committed:
Tests fixed

SCaptainCAP committed:
Bug fixes
